source/Calamari/Kubernetes/ContextProviders/CertificateAuthenticationContextProvider.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Text;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Kubernetes.Integration;
+
+namespace Calamari.Kubernetes.ContextProviders
+{
+    public class CertificateAuthenticationContextProvider
+    {
+        readonly IVariables variables;
+        readonly ILog log;
+        readonly IKubectl kubectl;
+
+        public CertificateAuthenticationContextProvider(IVariables variables,
+            ILog log,
+            IKubectl kubectl)
+        {
+            this.variables = variables;
+            this.log = log;
+            this.kubectl = kubectl;
+        }
+
+        public bool TrySetContext(
+            string @namespace,
+            string clusterUrl,
+            string clientCert,
+            string skipTlsVerification)
+        {
+            kubectl.ExecuteCommandAndAssertSuccess("config",
```

Comment:
All the others providers have a  `try-set` pattern where they first perform some test, and then perform a command to modify the context.
Why is this one different?

Reply:
I think the others have an obvious thing to check but this one doesn't. I guess we could just check all the required certs are there at the beginning. Probably worth adjusting 👍